crazyclock.ino

```diff
@@ -15,44 +16,57 @@ const char *ssid = "SSID";
 const char *password = "PASS";
 
 LocalDateTimeConverter plDateTimeConverter = LocalDateTimeConverter::PL;
-int mH, mM, mS;  // crazydata
-int tick = 1000; // initial value of tick =1s
-bool change;     // change of time
-bool isNtpAvailable;
 
-char formattedTimeBuffer[20] = "<initial value>";
-FakeTime fakeTimeFromRTC;
-FakeTime fakeTime;
+char formattedTimeBuffer[19] = "<initial value>";
 
-#if defined(ARDUINO_AVR_UNO) || defined(ARDUINO_AVR_NANO_EVERY)
-// Example for Arduino UNO with input signals on pin 2 and 3
-#define PIN_IN1 2
-#define PIN_IN2 3
-
-#elif defined(ESP8266)
 // Example for ESP8266 NodeMCU with input signals on pin D5 and D6
-#define PIN_IN1 12
-#define PIN_IN2 14
-#endif
+#define ROTARY_ENCODER_CLK D5
+#define ROTARY_ENCODER_DT D6
+#define ROTARY_ENCODER_BUTTON_PIN D7
 
-#define RESET_BUTTON_PIN 13
+// for rotary encoder switch active is low state
+Debouncer debouncer(ROTARY_ENCODER_BUTTON_PIN, 50, Debouncer::Active::L,
+                    Debouncer::DurationFrom::TRIGGER);
 
-// maybe myMillis should be a function returning the result?
 unsigned long myMillis;
 unsigned long epochSeconds;
+unsigned long localEpochSeconds;
+
+byte secondFromRtc;
+byte currentSecond;
+unsigned long newSecondStartedAtMillis;
+unsigned long currentMillis;
+
+unsigned long programStartedSeconds;
+unsigned long programStartedMillis;
 
 WiFiUDP ntpUDP;
 NTPClient timeClient(ntpUDP, "europe.pool.ntp.org");
 hd44780_I2Cexp lcd;
 const int LCD_COLS = 16;
 const int LCD_ROWS = 2;
-RotaryEncoder encoder(PIN_IN1, PIN_IN2, RotaryEncoder::LatchMode::TWO03);
+RotaryEncoder encoder(ROTARY_ENCODER_DT, ROTARY_ENCODER_CLK,
+                      RotaryEncoder::LatchMode::TWO03);
+int pos = 0;
+int buttonState = 0;
 DS3231 rtc;
-bool is12h;
-bool isPM;
 
 void setup() {
-  pinMode(RESET_BUTTON_PIN, INPUT);
+  pinMode(ROTARY_ENCODER_BUTTON_PIN, INPUT_PULLUP);
```

Comment:
Does it work now correctly?

Reply:
Yes, it works without additional resistor.